src/Microsoft.Extensions.ML/ServiceCollectionExtensions.cs

```diff
@@ -19,27 +20,50 @@ public static class ServiceCollectionExtensions
         /// <param name="services">
         /// The <see cref="IServiceCollection "/> to add services to.
         /// </param>
+        /// <param name="implementationFactory">
+        /// The factory that creates the service.
```

Comment:
(nit) "the service" is a little vague here. Would be better to say `that creates the <see cref="PredictionEnginePoolBuilder{TData, TPrediction}"/>`.

Reply:
I actually just copied this from the corresponding parameter in `AddSingleton` for consistency. Take a look there, and let me know if you still think it should be changed.

Reply:
In the context of `AddSingleton`, saying `the service` makes sense because it can be any kind of service. But here, we are in the context of `AddPredictionEnginePool`, so `the service` doesn't really fit. As a reader, I found myself asking the question "what service?", which is why I think replacing `service` with `PredictionEnginePoolBuilder` makes more sense. We are adding a `PredictionEnginePoolBuilder`, and the `implementationFactory` is the thing that makes it.

Reply:
OK, have updated.